### DISCUSSION TITLE
Executing a function for each element and correct typing of it

### DISCUSSION THREAD

**Discussion by Sanchoyzer** [python/typing#1270] [Reviews]
Let's look at this code:

```
import dataclasses

@dataclasses.dataclass
class MyClass:
    x: int
    y: int

    def __init__(self, **kwargs) -> None:
        any(object.__setattr__(self, k, v) for k, v in kwargs.items() if k in self.fields())

    @classmethod
    def fields(cls) -> set[str]:
        return {field.name for field in dataclasses.fields(cls)}


MyClass(**{'x': 1, 'y': 2, 'z': 3})
```

There is a mypy error:
```
error: "__setattr__" of "object" does not return a value
```

Of course, I can write it like a loop, but it'll take three lines instead of one.
What is the correct way (in one line) to execute a function (without returning a result value) for each element from a collection?

**Comment by sobolevn** [comment]
> but it'll take three lines instead of one

Lines are basically free ;)
Plus, this would be ideomatic usage.

The reported error is correct. You cannot use `object.__setattr__` with `any`, it does not make any sense.

**Reply by Sanchoyzer** [reply]
I see that mypy reported (but the code still executes successfully). My question: is there any way to do it in one line and respect the mypy request?

**Reply by JelleZijlstra** [reply]
`for k, v in [k, v for k, v in kwargs.items() if k in self.fields()]: object.__setattr__(self, k, v)`

**Reply by sobolevn** [reply]
🙈

**Reply by Sanchoyzer** [reply]
I can do it shorter, but still not elegant :)
```
for k in set(kwargs.keys()) & self.fields(): object.__setattr__(self, k, kwargs[k])
```

**Reply by AlexWaygood** [reply]
Dictionary keys are a set-like object, so the call to `set(kwargs.keys())` is redundant; you can golf it down further to:

```python
for k in kwargs.keys() & self.fields(): object.__setattr__(self, k, kwargs[k])
```

**Reply by jeremyn** [reply]
You can also one-line it in `__init__` if you create a new function `set_attributes(...)` and move the logic there. Or, have the caller filter `kwargs` before passing it to `__init__`. Also, translating between keys and attributes is something `types.SimpleNamespace` already does, maybe that could fit in somewhere.
